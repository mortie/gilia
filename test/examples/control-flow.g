print "If"
if 'true {print "true is true"} {print "true is not true"}
if 'false {print "false is true"} {print "false is not true"}
if 'true {print "true is true"}
if 'false {print "false is true"} # This should output nothing
print (if 'true {"If returns its true value"})
print (if 'false {0} {"If returns its false value"})

print "\nRun loop 10 times"
i := 0
loop {
	print "Hello World"
	i = i + 1
	if i >= 10 {'stop}
}

print "\nWith while this time"
i := 0
while {i < 10} {
	print "Hallo Wrodl"
	i = i + 1
}

print "\nAnd with the for loop"
i := 0
iter := {
	j := i
	i = i + 1
	if j < 10 {j} {'stop}
}
for iter {print "Hello with" $}

print "\nArray iterator"
array-iter := {
	arr := $.0
	idx := 0
	{
		if idx >= (len arr) {
			'stop
		} {
			j := idx
			idx = idx + 1
			[j arr.(j)]
		}
	}
}
for (array-iter ["hello" "world" "how" "are" "you" 10 20 'true]) print
