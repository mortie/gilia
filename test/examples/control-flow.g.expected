If
true is true
false is not true
true is true
If returns its true value
If returns its false value

Run loop 10 times
Hello World
Hello World
Hello World
Hello World
Hello World
Hello World
Hello World
Hello World
Hello World
Hello World

With while this time
Hallo Wrodl
Hallo Wrodl
Hallo Wrodl
Hallo Wrodl
Hallo Wrodl
Hallo Wrodl
Hallo Wrodl
Hallo Wrodl
Hallo Wrodl
Hallo Wrodl

And with the for loop
Hello with [0]
Hello with [1]
Hello with [2]
Hello with [3]
Hello with [4]
Hello with [5]
Hello with [6]
Hello with [7]
Hello with [8]
Hello with [9]

Array iterator
[0 hello]
[1 world]
[2 how]
[3 are]
[4 you]
[5 10]
[6 20]
[7 (true)]
